plugin_tests/cli_results_test.py

```diff
@@ -64,28 +65,56 @@ def _runTest(self, cli_args=(), cli_kwargs={}, outputs={}, contains=[],
                     file.
                 'contains': a list of phrases that must be present in the first
                     256 kb of the file.
+        :param in_process: a flag indicating whether to run the cli in-process.
+            Defaults to True, and when True enables coverage testing to reach
+            the CLIs.  When False, subprocess is used, which enables capturing
+            stdout and stderr.
         :param contains: a list of phrases that must be present in the stdout
-            output of the cli.
+            output of the cli.  Only valid if in_process is False.
         :param excludes: a list of phrases that must be not present in the
-            stdout output of the cli.
+            stdout output of the cli.  Only valid if in_process is False.
         :returns: stdout from the test.
+
         """
         chunkSize = 256 * 1024
 
-        stdout = ''
+        if in_process and (contains or excludes):
+            raise ValueError("contains and excludes parameters may only be used"
+                             " if in_process is False.")
+
         cwd = os.environ.get('CLI_CWD')
-        cmd = ['python', os.environ['CLI_LIST_ENTRYPOINT']]
         tmppath = tempfile.mkdtemp()
         try:
-            cmd += [arg if not arg.startswith('tmp_') else
-                    os.path.join(tmppath, arg) for arg in cli_args]
+            cmd = [arg if not arg.startswith('tmp_') else
+                   os.path.join(tmppath, arg) for arg in cli_args]
             cmd += ['--%s=%s' % (
                 k, v if not v.startswith('tmp_') else os.path.join(tmppath, v))
                 for k, v in six.iteritems(cli_kwargs)]
-            process = subprocess.Popen(
-                cmd, shell=False, stdout=subprocess.PIPE, cwd=cwd)
-            stdout, stderr = process.communicate()
-            self.assertEqual(process.returncode, 0)
+            if in_process:
```

Comment:
You don't need two separate code paths here -- you can use `runpy.run_path` *and* capture stdout and stderr.  Much like with `old_sys_argv`:
```
                stdout = None
                old_sys_stdout = sys.stdout
                old_sys_stderr = sys.stderr
                sys.stdout = stdout = six.BytesIO()
                sys.stderr = stderr = six.BytesIO()
                try:
...
                finally:
                    sys.argv[:] = old_sys_argv
                    sys.stdout = old_sys_stdout
                    sys.stderr = old_sys_stderr
                stdout = stdout.getvalue()
                stderr = stderr.getvalue()
```